src/plugins/execution-time-estimator-navgraph/navgraph_estimator.cpp

```diff
@@ -35,45 +35,47 @@ namespace fawkes {
 NavGraphEstimator::NavGraphEstimator(LockPtr<NavGraph>  navgraph,
                                      Configuration *    config,
                                      const std::string &cfg_prefix)
-: config_(config), cfg_prefix_(cfg_prefix), navgraph_(navgraph)
+: ExecutionTimeEstimator(config, cfg_prefix),
+  navgraph_(navgraph),
+  source_names_(config_, cfg_prefix_, "start", ""),
+  dest_names_(config_, cfg_prefix_, "target")
 {
 	last_pose_x_ = config->get_float_or_default("plugins/amcl/init_pose_x", 0);
 	last_pose_y_ = config->get_float_or_default("plugins/amcl/init_pose_y", 0);
-	speed_       = config->get_float_or_default((cfg_prefix_ + "speed").c_str(), 0.5);
 }
 
 bool
-NavGraphEstimator::can_execute(const Skill &skill) const
+NavGraphEstimator::can_provide_exec_time(const Skill &skill)
```

Comment:
As mentioned above, this can be kept `const` if the mutex is declared `mutable`

Reply:
Done in d100bffa49be6efa274fc1464d7613dedf209dda